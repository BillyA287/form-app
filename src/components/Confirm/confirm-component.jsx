import React from "react";
import Dialog from "@material-ui/core/Dialog";
import AppBar from "@material-ui/core/AppBar";
import { List, ListItem, ListItemText } from "@material-ui/core/";
import Button from "@material-ui/core/Button";


export class Confirm extends React.Component {
  continue = (event) => {
    event.preventDefault();
    // send form
    this.props.nextStep();
  };

  back = (event) => {
    event.preventDefault();
    this.props.previousStep();
  };
  render() {
    const {
      values: { firstName, lastName, email, occupation, city, bio },
    } = this.props;

    return (
      <>
        <Dialog open fullWidth maxWidth="sm">
          <AppBar title="Confirm User Data" />
          <List>
            <ListItem>
              <ListItemText
                data-test="firstName"
                primary="First Name"
                secondary={firstName}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                data-test="lastName"
                primary="Last Name"
                secondary={lastName}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                data-test="Email"
                primary="Email"
                secondary={email}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                data-test="Occupation"
                primary="Occupation"
                secondary={occupation}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                data-test="City"
                primary="City"
                secondary={city}
              />
            </ListItem>
            <ListItem>
              <ListItemText
                data-test="Bio"
                primary="Bio"
                secondary={bio}
              />
            </ListItem>
          </List>
          <br />

          <Button color="secondary" variant="contained" onClick={this.back}>
            Back
          </Button>

          <Button color="primary" variant="contained" onClick={this.continue}>
            Confirm & Continue
          </Button>
        </Dialog>
      </>
    );
  }
}

export default Confirm;
