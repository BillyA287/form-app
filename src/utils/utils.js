import React from 'react'




export const findTestbyAtrribute = (wrapper, attr) => {
     wrapper = wrapper.find(`[data-test='${attr}']`);
    return wrapper;
}