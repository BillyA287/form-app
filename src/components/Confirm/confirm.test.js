import React from 'react'
import Confirm from './confirm-component'
import {shallow} from 'enzyme'
import {findTestbyAtrribute} from '../../utils/utils'


// it("Correctly renders Confirm component", () => {

//   expect(shallow(<Confirm firstName={values.firstName} lastName={jest.fn()} email={jest.fn()} occupation={jest.fn()} city={jest.fn()} bio={jest.fn()}/>)).toMatchSnapshot();
// });


const setUp = (props={}) => {
    const component = shallow(<Confirm {...props} />);
    return component
}

describe('Confirm Component', ()=> {

    describe('props passed in', ()=> {

        it('should render without errors', ()=> {
            const props = {
              firstName: "b",
              lastName: "a",
              email: "@",
              occupation: "g",
              city: "d",
              bio: "t",
            };
            const component = shallow(<Confirm values={props} />);
            expect(component.length).toMatchSnapshot()
        })
    })

    describe('no props passed in', ()=> {
        let wrapper;
        beforeEach(()=>{
            wrapper = setUp()
        })
    })
})