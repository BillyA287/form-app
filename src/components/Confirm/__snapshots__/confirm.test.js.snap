// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Component props passed in should render without errors 1`] = `1`;
